### PR TITLE
Feature/upload vagrant box

### PR DESCRIPTION
**Pull request by changev**
Seprate upload to atlas from packer templates.

This script can upload all box files under a dir to atlas.

@panpan0000 @PengTian0

### CODE REVIEW

**Comment by anhou** [comment]
@changev confilcts exists, needs to resolve.

**Comment by changev** [comment]
@panpan0000 @PengTian0 please review, thanks

**Comment by PengTian0** [comment]
:+1: After fix the comment.

File: manifest-build-tools/application/release_to_atlas.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+            return True
+        print "{0} provider of version {1} doesn't' exist, will be created soon".format(provider, atlas_version)
+        return False
+
+    def generate_url(self, purpose, atlas_version=None, provider=None):
+        """
+        Tool method, Generate all possible urls according to purpose
+        """
+        purpose_handler = {
+            "check_version": lambda atlas_version, provider: "/".join([self.atlas_url, self.box, "version/{0}".format(atlas_version)]),
```
Comment (panpan0000) [inline manifest-build-tools/application/release_to_atlas.py]: Woo!!!

File: manifest-build-tools/application/release_to_atlas.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    parser.add_argument('--atlas-name',
+                        help="The repo name under a specific account of atlas, default: rackhd",
+                        action='store')
+
+    parser.add_argument('--atlas-token',
+                        help="atlas access token",
+                        action='store')
+
+    parser.add_argument('--atlas-version',
+                        help="atlas access token",
```
Comment (panpan0000) [inline manifest-build-tools/application/release_to_atlas.py]: wrong "help"

File: manifest-build-tools/application/release_to_atlas.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+    box_files = common.find_specify_type_files(build_directory, ".box", depth=1)
+    if len(box_files) == 0:
+        print "No box found under {0}".format(build_directory)
+
+    for full_file_path in box_files:
+        if not atlas_version:
+            if is_release:
+                # Box file name is like "rackhd-ubuntu-14.04-1.2.3-20161207024UTC.box"
+                # Extract 1.2.3-20161207024UTC only
```
Comment (panpan0000) [inline manifest-build-tools/application/release_to_atlas.py]: good job to handle case when "atlas_version" is blank.
if "is_release", why still use long name 1.2.3-20161207024UTC ?  it's a nightly build version ?

File: manifest-build-tools/application/release_to_atlas.py [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+                        required=True,
+                        help="A directory where box files laid in",
+                        action='store')
+
+    parser.add_argument('--atlas-url',
+                        help="Base URL for Atlas, default: https://atlas.hashicorp.com/api/v1",
+                        action='store')
+
+    parser.add_argument('--atlas-username',
+                        help="The account name of atlas, default: rackhd",
```
Comment (panpan0000) [inline manifest-build-tools/application/release_to_atlas.py]: why not "required=True"

Comment (changev) [inline manifest-build-tools/application/release_to_atlas.py]: Forgotten that, I'll fix this.

File: manifest-build-tools/application/release_to_atlas.py (hunk 1 above)
Comment (changev) [inline manifest-build-tools/application/release_to_atlas.py]: First I think use a ```dict``` is more clear than ```swith case```, but at last I find this style(anonymous function in python) is complex too....
So be it.

File: manifest-build-tools/application/release_to_atlas.py (hunk 3 above)
Comment (changev) [inline manifest-build-tools/application/release_to_atlas.py]: Not a long name 1.2.3-20161207024UTC, but just 0.12.07.

File: manifest-build-tools/application/release_to_atlas.py (hunk 2 above)
Comment (changev) [inline manifest-build-tools/application/release_to_atlas.py]: Got it

File: manifest-build-tools/application/release_to_atlas.py (hunk 3 above)
Comment (changev) [inline manifest-build-tools/application/release_to_atlas.py]: Not a long name, but just 0.12.07 for short.
The logic in this snippet is a little chaotic.
I'll modify this to:
* if is_release, atlas_version must be needed.
* if not release, will extract atlas version 0.12.07 from 1.2.3-20161207024UTC
* all other conditions will exit 1 with error.

**Comment by changev** [comment]
@panpan0000 @PengTian0 
PR has been updated, please review, thanks.

**Comment by panpan0000** [comment]
:+1:

**Comment by PengTian0** [comment]
:+1:

**Pull request merged**